core/roslib/src/roslib/packages.py

```diff
@@ -402,8 +402,13 @@ def find_node(pkg, node_type, rospack=None):
 def _executable_filter(test_path):
     s = os.stat(test_path)
     flags = stat.S_IRUSR | stat.S_IXUSR
-    if os.name == 'nt' and os.path.splitext(test_path)[1] == '.py':
-        flags = stat.S_IRUSR
+    if os.name == 'nt':
+        _, ext = os.path.splitext(test_path)
+        # python scripts in ROS tend to omit .py extension since they could become executable
+        # by adding a shebang line (#!/usr/bin/env python) in linux environment
+        # special handle this case in Windows environment
+        if ext.lower() in ['.py', '']:
```

Comment:
Is the `.lower` really necessary? I have never encountered `.PY` as a file extension.

Reply:
we think it wouldn't hurt, there is no guarantee whether anyone would actually use `.PY` as the extension. Meantime, `.lower` should be a quite simple implementation, so it should not hurt complexity or efficiency.